# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/nsICookieManager.idl
# Commit: 123069093c2f
# Full Hash: 123069093c2fc18e87e9ee19c2d22815bc6b2804
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2025-03-28 21:29:38
# Regressor Bug: 1951750
# File Overlap Count: 4
# Description:
#   Bug 1951750 - Handle domain argument in CookieStoreParent r=baku,edgul,cookie-reviewers
#   
#   Depends on D241966
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D240695
# ==============================================================================

diff -r f3959de161ba -r 123069093c2f netwerk/cookie/nsICookieManager.idl
--- a/netwerk/cookie/nsICookieManager.idl	Thu Mar 27 19:06:33 2025 +0000
+++ b/netwerk/cookie/nsICookieManager.idl	Thu Mar 27 19:06:33 2025 +0000
@@ -6,14 +6,20 @@
 #include "nsISupports.idl"
 #include "nsICookie.idl"
 #include "nsIThirdPartyCookieBlockingExceptionListService.idl"
+#include "nsIURI.idl"
 
 %{ C++
+#include <functional>
 namespace mozilla {
 class OriginAttributes;
+namespace net {
+class CookieStruct;
+} // namespace net
 } // mozilla namespace
 %}
 
 [ptr] native OriginAttributesPtr(mozilla::OriginAttributes);
+native CheckStructFunctionRef(const std::function< bool(mozilla::net::CookieStruct&) > &);
 
 /**
  * An optional interface for accessing or removing the cookies
@@ -139,7 +145,8 @@
            [optional] in boolean aIsPartitioned);
 
   [notxpcom]
-  nsresult addNative(in AUTF8String aHost,
+  nsresult addNative(in nsIURI aCookieURI,
+                     in AUTF8String aHost,
                      in AUTF8String aPath,
                      in ACString    aName,
                      in AUTF8String aValue,
@@ -151,7 +158,8 @@
                      in int32_t aSameSite,
                      in nsICookie_schemeType aSchemeMap,
                      in boolean     aIsPartitioned,
-                     in nsIDPtr     aOperationID);
+                     in nsIDPtr     aOperationID,
+                     in CheckStructFunctionRef aCheckValid);
 
   /**
    * Find whether a given cookie already exists.